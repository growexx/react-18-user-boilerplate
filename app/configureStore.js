import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
