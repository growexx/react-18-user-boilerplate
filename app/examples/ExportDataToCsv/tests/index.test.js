/**
 *
 * Tests for ExportDataToCsv
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import history from 'utils/history';
import { store } from 'configureStore';
import ExportDataToCsv from '../index';
let globalStore;
// if not using firebase messaging remove this mock
jest.mock('firebase/messaging', () => {
  const actualModule = jest.requireActual('firebase/messaging');
  return {
    ...actualModule,
    onMessage: jest.fn(),
    getMessaging: jest.fn(),
  };
});
const componentWrapper = () =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <ExportDataToCsv />
        </Router>
      </IntlProvider>
    </Provider>,
  );

describe('<ExportDataToCsv />', () => {
  beforeAll(() => {
    globalStore = store;
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should trigger export', () => {
    const { container } = componentWrapper();
    const checkbox = container.querySelector('input');
    const element = checkbox;
    fireEvent.click(element);
    const { getAllByTestId } = componentWrapper();
    const buttonElement = getAllByTestId('ExportButton')[0];
    fireEvent.click(buttonElement);
    expect(buttonElement.tagName).toEqual('BUTTON');
  });
});
