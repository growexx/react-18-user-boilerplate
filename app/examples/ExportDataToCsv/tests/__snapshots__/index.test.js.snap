// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportDataToCsv /> Should render and match the snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c0 button {
  margin-bottom: 12px;
}

.c1 {
  align-self: flex-end;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-190m0jy ant-btn-primary"
        data-testid="ExportButton"
        disabled=""
        type="button"
      >
        Export Data
      </button>
    </div>
    <div
      class="ant-table-wrapper css-dev-only-do-not-override-190m0jy"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-190m0jy"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-scroll-horizontal"
            data-testid="DataTable"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
                style="overflow-x: auto; overflow-y: hidden;"
              >
                <table
                  style="width: 350px; min-width: 100%; table-layout: auto;"
                >
                  <colgroup>
                    <col
                      class="ant-table-selection-col"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-selection-column"
                        scope="col"
                      >
                        <div
                          class="ant-table-selection"
                        >
                          <label
                            class="ant-checkbox-wrapper css-dev-only-do-not-override-190m0jy"
                          >
                            <span
                              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-190m0jy"
                            >
                              <input
                                aria-label="Select all"
                                class="ant-checkbox-input"
                                type="checkbox"
                              />
                              <span
                                class="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </div>
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Name
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Age
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Address
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      class="ant-table-measure-row"
                      style="height: 0px; font-size: 0px;"
                    >
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-190m0jy"
                        >
                          <span
                            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-190m0jy"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        John Brown
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        New York No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-190m0jy"
                        >
                          <span
                            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-190m0jy"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Jim Green
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        42
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        London No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="3"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-190m0jy"
                        >
                          <span
                            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-190m0jy"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Joe Black
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Sidney No. 1 Lake Park
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
